astroid/node_classes.py

```diff
@@ -630,12 +631,44 @@ def nodes_of_class(self, klass, skip_klass=None):
         """
         if isinstance(self, klass):
             yield self
+
+        if skip_klass is None:
+            for child_node in self.get_children():
+                for matching in child_node.nodes_of_class(klass, skip_klass):
+                    yield matching
+
+            return
+
         for child_node in self.get_children():
-            if skip_klass is not None and isinstance(child_node, skip_klass):
+            if isinstance(child_node, skip_klass):
                 continue
             for matching in child_node.nodes_of_class(klass, skip_klass):
                 yield matching
 
+    def get_assign_nodes(self):
+        for child_node in self.get_children():
+            for matching in child_node.get_assign_nodes():
```

Comment:
You can use `yield from` now that we're using Python 3 only (unless you want to backport this to the 1.6 branch). That actually might give a speedup